core/templates/dev/head/domain/exploration/EditableExplorationBackendApiService.js

```diff
@@ -0,0 +1,163 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to send changes to a exploration to the backend.
+ */
+
+oppia.factory('EditableExplorationBackendApiService', [
+  '$http', '$q', 'EXPLORATION_DATA_URL_TEMPLATE',
+  'EDITABLE_EXPLORATION_DATA_URL_TEMPLATE', 'UrlInterpolationService',
+  'ReadOnlyExplorationBackendApiService',
+  function($http, $q, EXPLORATION_DATA_URL_TEMPLATE,
+    EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, UrlInterpolationService,
+    ReadOnlyExplorationBackendApiService) {
+    var _fetchExploration = function(
+        explorationId, applyDraft, v, successCallback, errorCallback) {
+      var explorationDataUrl = UrlInterpolationService.interpolateUrl(
+        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+          exploration_id: String(explorationId)
+        });
+
+      params = {};
+      if (applyDraft) {
+        params.apply_draft = applyDraft;
+      }
+      if (v || v === 0) {
+        params.v = v;
+      }
+
+      $http.get(explorationDataUrl, {
+        params: params
+      }).then(function(response) {
+        var exploration = angular.copy(response.data);
+        if (successCallback) {
+          successCallback(exploration);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    var _updateExploration = function(
+        explorationId, explorationVersion, commitMessage, changeList,
+        successCallback, errorCallback) {
+      var editableExplorationDataUrl = (
+        UrlInterpolationService.interpolateUrl(
+        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+          exploration_id: explorationId
+        })
+      );
+
+      var putData = {
+        version: explorationVersion,
+        commit_message: commitMessage,
+        change_list: changeList
+      };
+      $http.put(editableExplorationDataUrl, putData).then(function(response) {
+        // The returned data is an updated exploration dict.
+        var exploration = angular.copy(response.data);
+
+        // Update the ReadOnlyExplorationBackendApiService's cache with the
+        // new exploration.
+        ReadOnlyExplorationBackendApiService.cacheExploration(
+          explorationId, exploration);
+
+        if (successCallback) {
+          successCallback(exploration);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    var _deleteExploration = function(
+        explorationId, role, successCallback, errorCallback) {
+      var editableExplorationDataUrl = (
+        UrlInterpolationService.interpolateUrl(
+        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+          exploration_id: explorationId
+        })
+      );
+
+      params = {};
+      if (role) {
+        params.role = JSON.stringify(role);
```

Comment:
Ping. :) We did this with version, so we should also be able to do it with ``role``, right?